kip-0017.md

```diff
@@ -8,105 +8,425 @@ Category: Interface
 
-kadena:mainnet01:22ddc64851718e9d41d98b0f33d5e328ae5bbbbd97aed9885adac0f2d070ff9c
+These Wallet Connect accounts are retrieved from the `accounts` property of a [Paring response](#pairing-with-walletconnect).
 
-kadena:testnet04:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89
+#### Example kadena_getAccounts_v1 Request and Response
+
+```jsonc
+// kadena_getAccounts_v1 Method Request
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "kadena_getAccounts_v1",
+  "params": {
+    "accounts": [
+      {
+        "account": "kadena:mainnet01:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+        "contracts": ["coin"] // optional, when omitted, the wallet returns all known fungible accounts
+      }
+    ]
+  }
+}
+````
+
+```jsonc
+// kadena_getAccounts_v1 Method Response
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": [
+    {
+      "accounts": [
+        {
+          "account": "kadena:mainnet01:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+          "publicKey": "38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+          "kadenaAccounts": [
+            {
+              "name": "w:aoriestnaoirsetnaorisetn",
+              "contract": "coin",
+              "chains": ["0", "1"]
+            },
+            {
+              "name": "k:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+              "contract": "coin",
+              "chains": [
+                "0",
+                "1",
+                "2",
+                "3",
+                "4",
+                "5",
+                "6",
+                "7",
+                "8",
+                "9",
+                "10",
+                "11",
+                "12",
+                "13",
+                "14",
+                "15",
+                "16",
+                "17",
+                "18",
+                "19"
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
 ```
 
-## Default Methods
+#### kadena_getAccounts_v1 Request
 
-- The supported methods are:
+This method needs to be called with a `params` object with the following properties:
+
+1. `params`: `Object`  
+    1.1. `accounts`: `Array<Object>` -  
+      1.1.1 `account`: `string` - the public key (a Wallet Connect [Account](#wallet-connect-accounts)) for which the Kadena account name is requested.   
+      1.1.2 `contracts`: `Array<string>` (optional) - array of fungible contracts for which the Kadena account name is requested. Returns all known Kadena accounts if omitted.  
+
+#### kadena_getAccounts_v1 Response
 
+The method response contains a `result` object that contains the following properties:
+
+1. `result`: `Object`  
+    1.1. `accounts`: `Array<Object>`  
+    __ 1.1.1 `account`: `string` - the Wallet Connect [Account](#wallet-connect-accounts) requested.  
```

Comment:
The __ are my attempt to get the tabs to work in the preview. Didn't add it to the rest in case we wanted to go with a different approach

Reply:
I like this approach, in combination with newlines it looks good in the rendered version now.